lesson-2

eliseevdry opened the pull request.

eliseevdry committed:
lesson-2

eliseevdry committed:
add Idempotency-Key header in post methods

eliseevdry committed:
fix

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+          name: likeCountGte
+          schema:
+            type: integer
+          description: Фильтр по количеству лайков (больше или равно)
```
А зачем нужен этот фильтр?

eliseevdry replied:
ну, думал для поиска по количеству лайков, но наверное это лучше решать сортировкой (вряд ли будет такое что пользователь захочет видеть посты в определенном диапазоне лайков)

eliseevdry replied:
убрал

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+          name: likeCountLt
+          schema:
+            type: integer
+          description: Фильтр по количеству лайков (меньше)
```
Аналогично

eliseevdry replied:
убрал

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+          name: commentsCountGte
+          schema:
+            type: integer
+          description: Фильтр по количеству комментариев (больше или равно)
```
Аналогично

eliseevdry replied:
убрал

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+          name: commentsCountLte
+          schema:
+            type: integer
+          description: Фильтр по количеству комментариев (меньше или равно)
```
Аналогично

eliseevdry replied:
убрал

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+            type: string
+            enum: [title, latitude, longitude, country, city, likeCount, commentsCount, createdDate]
+            default: createdDate
+          description: Поле для сортировки
```
Я бы не давал кастомизация в этом случае - делал бы обратный хронологический порядок и все

eliseevdry replied:
просто там требование было вроде про популярные места (ну, вообще, думаю можно сортировку по комментам и лайкам делать без ведома пользователя)

eliseevdry replied:
убрал сортировку

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+  /comments:
+    post:
+      summary: Добавить комментарий к посту
+      description: Добавляет комментарий к указанному посту
```
"Указанный пост" нужно указывать в URI

eliseevdry replied:
добавил в queryParam

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+
+    get:
+      summary: Получить список комментариев с пагинацией
+      description: Возвращает список комментариев для поста
```
Аналогично, я бы указывал пост в URI

eliseevdry replied:
добавил в queryParam

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+        '500':
+          description: Внутренняя ошибка сервера
+
+  /likes:
```
Аналогично про указывание поста

eliseevdry replied:
добавил в queryParam

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+        '500':
+          description: Внутренняя ошибка сервера
+
+  /follow:
```
В REST е принято указывать глаголы в URI

eliseevdry replied:
исправил на /subscriptions

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -0,0 +1,519 @@
+          type: number
+          format: float
+          example: 2.2945
+        content:
```
Почему блобы идут вместе с постом? Обычно делают двухэтапную загрузку, сначала грузят картинки, получают урлы и затем линкуют их к посту

eliseevdry replied:
думал делать это разделение на сервере, в одной транзакции. То есть если не получилось сохранить картинку в BLOB storage, то мы пост просто не сохраняем и 503 ошибку отдаем. А если получилось, то идем дальше по транзакции: сохраняем url в DTO.
Вижу 2 плюса такого подхода:
1) меньше задержка на сетевое взаимодействие (если BLOB storage ближе находится к серверу, чем к клиенту)
2) клиенту не нужно думать о логике удаления картинки, если транзакция завершилась неудачей

VladimirBalun replied:
Я бы придерживался стандартного подхода

eliseevdry replied:
исправил, добавил POST /media

eliseevdry committed:
fix review

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -363,6 +319,13 @@ paths:
       security:
         - BearerAuth: []
       parameters:
+        - in: query
```
Это не в URI

eliseevdry replied:
исправил, теперь для likes вид `/posts/{postId}/likes` и для comments вид `/posts/{postId}/comments`

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -394,6 +357,13 @@ paths:
       security:
         - BearerAuth: []
       parameters:
+        - in: query
```
Это не в URI

eliseevdry replied:
исправил, теперь для likes вид `/posts/{postId}/likes` и для comments вид `/posts/{postId}/comments`

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -459,6 +436,13 @@ paths:
       security:
         - BearerAuth: []
       parameters:
+        - in: query
```
Это не в URI

eliseevdry replied:
исправил, теперь для likes вид `/posts/{postId}/likes` и для comments вид `/posts/{postId}/comments`

VladimirBalun commented on api/rest_api.yml:
```diff
@@ -428,6 +398,13 @@ paths:
       security:
         - BearerAuth: []
       parameters:
+        - in: query
```
Это не в URI

eliseevdry replied:
исправил, теперь для likes вид `/posts/{postId}/likes` и для comments вид `/posts/{postId}/comments`

eliseevdry committed:
fix review - 2

Добавил:
1. POST /media для добавления медиа-фалов и получение url
2. добавил {postId} в uri

Так же добавил:
1. Методы редактирования и удаления поста и комментария
2. Код ошибки 403 - когда нет прав на редактирование / удаление сущностей
3. Теги для удобства
